src/LocalDateRange.php

```diff
@@ -201,7 +201,7 @@ public function getIntersectionWith(LocalDateRange $that) : LocalDateRange
     /**
      * Returns an iterator for all the dates contained in this range.
      *
-     * @return LocalDate[]
+     * @return \Generator<LocalDate>
```

Comment:
```suggestion
     * @return \Iterator<LocalDate>
```
Let's not make the contract more specific than it needs to be.